synapse_antispam/mjolnir/antispam.py

```diff
@@ -127,3 +136,24 @@ def user_may_publish_room(self, user_id, room_id):
     @staticmethod
     def parse_config(config):
         return config  # no parsing needed
+
+# New module API
+class Module(object):    
+    def __init__(self, config, api):
+        self.antispam = AntiSpam(config, api)        
+        self.api.register_spam_checker_callbacks(
+            check_event_for_spam=self.antispam.check_event_for_spam,
+            user_may_invite=self.antispam.user_may_invite,
+            check_username_for_spam=self.antispam.check_username_for_spam,
+        )
+
+    # Spam-checker API
+    # Note that these are `async`, by opposition to the APIs in `AntiSpam`.
+    async def check_event_for_spam(self, event):
+        return self.antispam.check_event_for_spam(self, event)
+
+    async def user_may_invite(self, inviter_user_id, invitee_user_id, room_id):
```

Comment:
recommend including type annotations on new methods:

```suggestion
    async def user_may_invite(self, inviter_user_id: str, invitee_user_id: str, room_id: str) -> bool:
```